kolibri/core/deviceadmin/tests/test_dbrestore.py

```diff
@@ -0,0 +1,235 @@
+        'NAME': os.path.join(tempfile.mkdtemp(), "test{}.db".format(random.randint(0, 100000))),
+        'OPTIONS': {
+            'timeout': 100,
+        }
+    }
+}
+
+
+def is_sqlite_settings():
+    """
+    This does not work during pytest collection, needs to be called while
+    executing tests!
+    """
+    return 'sqlite3' in settings.DATABASES['default']['ENGINE']
+
+
+def mock_status_not_running():
+    raise NotRunning(STATUS_UNKNOWN)
+
+
+def test_latest():
+
+    with pytest.raises(RuntimeError):
+        call_command("dbrestore", latest=True)
+
+
+def test_illegal_command():
+
+    with pytest.raises(CommandError):
+        call_command("dbrestore", latest=True, dump_file="wup wup")
+
+
+def test_no_restore_from_no_file():
+
+    with pytest.raises(CommandError):
+        call_command("dbrestore", dump_file="does not exist")
+
+
+def test_active_kolibri():
+    """
+    Tests that we cannot restore while kolibri is active
+    """
+
+    with patch(
+        "kolibri.utils.server.get_status",
+        return_value=(12345, "http://127.0.0.1", 1234)
+    ) as gs:
+        with pytest.raises(SystemExit):
+            call_command("dbrestore", latest=True)
+            gs.assert_called_once()
+
+
+def test_inactive_kolibri():
+    """
+    Tests that we cannot restore while kolibri is active
+    """
+
+    with patch(
+        "kolibri.utils.server.get_status",
+        side_effect=mock_status_not_running
+    ) as gs:
+        # Since there's no backups available during a test, this should fail!
+        with pytest.raises(RuntimeError):
+            call_command("dbrestore", latest=True)
+            gs.assert_called_once()
+
+
+def test_not_sqlite():
+    if is_sqlite_settings():
+        return
+    with pytest.raises(IncompatibleDatabase):
+        dbrestore("/doesnt/matter.file")
+
+
+def test_fail_on_unknown_file():
+    with pytest.raises(ValueError):
+        get_dtm_from_backup_name("this-file-has-no-time")
+
+
+@pytest.mark.django_db
+@pytest.mark.filterwarnings('ignore:Overriding setting DATABASES')
+def test_restore_from_latest():
+    """
+    Tests that we cannot restore while kolibri is active
+    """
+    if not is_sqlite_settings():
+        return
+    with patch(
+        "kolibri.utils.server.get_status",
+        side_effect=mock_status_not_running
+    ):
+        # Create something special in the database!
+        from kolibri.auth.models import Facility
+        Facility.objects.create(name="test latest", kind=FACILITY)
+        # Create a backup file from the current test database
+        call_command("dbbackup")
+
+        # Also add in a file with an old time stamp to ensure its ignored
+        sql = "syntax error;"
+        fbroken = "db-v{}_2015-08-02_00-00-00.dump".format(kolibri.__version__)
+        open(os.path.join(default_backup_folder(), fbroken), "w").write(sql)
+
+        # Add an unparsable file name
+        fbroken = "db-v{}_.dump".format(kolibri.__version__)
```

Comment:
this is inserting `\n` in the file name:

variable:

```
>>> kolibri.__version__
'0.5.0\n'
```

terminal:

```
~/Downloads $ cd ~/.kolibri/backups/
~/.k/backups $ ll
total 504
-rw-r-----  1 d  staff    42K Oct 16 16:45 db-v0.5.0?_2017-10-16_16-45-23.dump
```

finder:

![image](https://user-images.githubusercontent.com/2367265/31640711-c199dc6a-b294-11e7-8773-01373df4bc39.png)